### PR TITLE
enh(perf): export event logs

### PR DESCRIPTION
**Pull request by TamazC**
## Description

This PR aims to fix event logs extraction issue

**Fixes** 
# MON-6585

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.04.x
- [ ] 21.10.x
- [ ] 22.04.x
- [x] 22.10.x (master)

<h2> How this pull request can be tested ? </h2>

On the event log page click to "export csv" button.
Export should be work even with the huge data

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by a-launois — DISMISSED** [review]

File: www/include/eventLogs/export/Request.php [hunk 1]
```diff
@@ -0,0 +1,634 @@
...
+     * @param int|null $is_admin
+     * @return void
+     */
+    public function setIsAdmin(?int $is_admin): void
+    {
+        $this->is_admin = $is_admin;
+    }
+
+    /**
+     * @param array $lca
```
Comment (jeremyjaouen) [inline www/include/eventLogs/export/Request.php]: type of array should be specified

File: www/include/eventLogs/export/Formatter.php [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+
+    /**
+     * Generates formatted CSV  data
+     * @param iterable $logs
+     * @return iterable
+     */
+    public function formatLogs(iterable $logs): iterable
+    {
+        foreach ($logs as $log) {
+            usleep(300);
```
Comment (jeremyjaouen) [inline www/include/eventLogs/export/Formatter.php]: is usleep still needed ?

**Review by jeremyjaouen — CHANGES_REQUESTED** [review]

File: www/include/eventLogs/export/QueryGenerator.php [hunk 3]
```diff
@@ -0,0 +1,586 @@
...
+        /*
+         * Add Host
+         */
+        $str_unitH = "";
+        $str_unitH_append = "";
+        $host_search_sql = "";
+        if (count($this->tabHostIds) == 0 && count($this->tabSvc) == 0) {
+            if ($this->engine == "false") {
+                $req .= " AND `msg_type` NOT IN ('4','5') ";
+                $req .= " AND logs.host_name NOT LIKE '_Module_BAM%' ";
```
Comment (kduret) [inline www/include/eventLogs/export/QueryGenerator.php]:
```suggestion
                $req .= " AND logs.host_name NOT LIKE '\\_Module\\_BAM%' ";

```

File: www/include/eventLogs/export/Request.php (hunk 1 above)
Comment (TamazC) [reply www/include/eventLogs/export/Request.php]: there is no typed object in this array.

File: www/include/eventLogs/export/Request.php [hunk 4]
```diff
@@ -0,0 +1,634 @@
...
+        $end = time();
+
+        if ($this->endDate != '') {
+            $end = $this->dateStringToTimestamp($this->endDate);
+        }
+
+        return $end;
+    }
+
+    private function dateStringToTimestamp(string $dateString): int
```
[the remark this answers is not in the document]
Comment (TamazC) [reply www/include/eventLogs/export/Request.php:159]: 👍

**Review by kduret — DISMISSED** [review]

File: www/include/eventLogs/export/Formatter.php [hunk 5]
```diff
@@ -0,0 +1,401 @@
...
+    private string $down = '';
+    private string $unreachable = '';
+    private string $ok = '';
+    private string $warning = '';
+    private string $critical = '';
+    private string $unknown = '';
+
+    /**
+     * @param int $start
+     * @return void
```
Comment (jeremyjaouen) [inline www/include/eventLogs/export/Formatter.php:64]: no need to specify @return void in phpdoc

File: www/include/eventLogs/export/Formatter.php [hunk 6]
```diff
@@ -0,0 +1,401 @@
...
+     * @param string $unknown
+     * @return void
+     */
+    public function setUnknown(string $unknown): void
+    {
+        $this->unknown = $unknown;
+    }
+
+    /**
+     * @param array $hosts
```
Comment (jeremyjaouen) [inline www/include/eventLogs/export/Formatter.php]:
```suggestion
     * @param array $hosts
```
specify the type of data in the array

File: www/include/eventLogs/export/Formatter.php [hunk 7]
```diff
@@ -0,0 +1,401 @@
...
+{
+    private const DATE_FORMAT = 'Y/m/d';
+    private const TIME_FORMAT = 'H:i:s';
+    private const DATE_TIME_FORMAT = 'Y/m/d (H:i:s)';
+    private const SERVICE_ACKNOWLEDGEMENT_MSG_TYPE = 10;
+    private const HOST_ACKNOWLEDGEMENT_MSG_TYPE = 11;
+    private const ACKNOWLEDGMENT_MESSAGE_TYPE = 'ACK';
+    private const INITIAL_STATE_VALUE = 'INITIAL STATE';
+    private const NOTIFICATION_TYPE_VALUE = 'NOTIF';
+    private array $hosts = [];
```
Comment (jeremyjaouen) [inline www/include/eventLogs/export/Formatter.php:36]: specifiy the type of data in the array

Comment (TamazC) [reply www/include/eventLogs/export/Formatter.php:36]: done

**Review by jeremyjaouen — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Pull request merged**